crates/typst-pdf/src/lib.rs

```diff
@@ -253,12 +254,25 @@ fn write_catalog(ctx: &mut PdfContext, ident: Option<&str>, timestamp: Option<Da
         .pair(Name(b"Type"), Name(b"Metadata"))
         .pair(Name(b"Subtype"), Name(b"XML"));
 
+    let destinations = write_and_collect_destinations(ctx);
+
     // Write the document catalog.
     let mut catalog = ctx.pdf.catalog(ctx.alloc.bump());
     catalog.pages(ctx.page_tree_ref);
     catalog.viewer_preferences().direction(dir);
     catalog.metadata(meta_ref);
 
+    // Write the named destinations.
+    let mut name_tree = catalog.names();
```

Comment:
This is not the name tree yet - it is the name dictionary. You create the name tree using `name_tree.destinations`.